zerver/management/commands/deliver_scheduled_messages.py

```diff
@@ -49,6 +50,8 @@ def construct_message(self, scheduled_message: ScheduledMessage) -> Dict[str, An
             realm_id=scheduled_message.realm.id,
             content=scheduled_message.content,
         )
+        if scheduled_message.stream is not None and mention_data.message_has_wildcards():
+            check_wildcard_mention_restrictions(original_sender, scheduled_message.stream)
```

Comment:
If you look at `zerver/lib/mention.py`, you'll see that `possible_mentions` (which populates MentionData) just looks for mention-like syntax; it doesn't check for actual mentions.  That latter bit can only be done with actual markdown rendering.

I think the right fix is that we should move the `render_incoming_message` call from `do_send_messages` to `check_message`, so that we can access the actual mention details inside `check_message`.

@showell FYI in case you want to pick up this refactor and/or collaborate on it.